http-client/src/main/java/io/airlift/http/client/FullJsonResponseHandler.java

```diff
@@ -63,7 +68,7 @@ private FullJsonResponseHandler(JsonCodec<T> jsonCodec)
     {
         byte[] bytes = readResponseBytes(response);
         String contentType = response.getHeader(CONTENT_TYPE);
-        if ((contentType == null) || !MediaType.parse(contentType).is(MEDIA_TYPE_JSON)) {
+        if ((contentType == null) || !SUPPORTED_MEDIA_TYPES.contains(MediaType.parse(contentType))) {
```

Comment:
This doesn't seem right. The `JsonCodec` in this handler supports either JSON or Smile, but not both, so we shouldn't accept both. Also, the error handling logic and `getJson()` API (used by error handling) assumes JSON. It wouldn't work for Smile which is binary.